components/AffirmationCarousel.tsx

```diff
@@ -139,6 +142,47 @@ export function AffirmationCarousel({
     e.preventDefault();
   };
 
+  const computeAllFontSizes = () => {
+    if (!containerRef.current) return;
+
+    const containerWidth = containerRef.current.offsetWidth;
+    const containerHeight = containerRef.current.offsetHeight;
+
+    const newFontSizes = slides.map((text) => {
+      const charCount = Math.max(text.length, 1);
+
+      const availableHeight = containerHeight * 0.7;
+
+      const availableWidth = containerWidth * 0.9;
+
+      let minSize = 16;
+      let maxSize = containerHeight * 0.7;
+      let bestSize = minSize;
+      while (maxSize - minSize > 1) {
```

Comment:
This approach seems like it would be slower than just letting CSS (or even SVG) handle the resizing for us.

What functionality does this approach give us that the others can't, and are we sure it's worth it?